drivers/sensor/adc_thermistor/adc_thermistor.c

```diff
@@ -0,0 +1,181 @@
+/*
+ * Copyright (c) 2023 Basalte bv
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/device.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/adc.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(adc_thermistor, CONFIG_SENSOR_LOG_LEVEL);
+
+#define DT_DRV_COMPAT zephyr_adc_thermistor
+
+struct adc_thermistor_config {
+	const struct adc_dt_spec adc;
+
+	const int32_t *lut;
+	size_t lut_size;
+};
+
+struct adc_thermistor_data {
+	struct k_mutex mutex;
+
+	int32_t val;
+};
+
+static int adc_thermistor_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	const struct adc_thermistor_config *config = dev->config;
+	struct adc_thermistor_data *data = dev->data;
+	int ret;
+
+	uint16_t buf;
+	struct adc_sequence sequence = {
+		.buffer = &buf,
+		/* buffer size in bytes, not number of samples */
+		.buffer_size = sizeof(buf),
+	};
+
+	/* Allow fetching using ambient or die temperature channel */
+	switch (chan) {
+	case SENSOR_CHAN_ALL:
+	case SENSOR_CHAN_AMBIENT_TEMP:
+	case SENSOR_CHAN_DIE_TEMP:
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	k_mutex_lock(&data->mutex, K_FOREVER);
+
+	ret = adc_channel_setup_dt(&config->adc);
+	if (ret < 0) {
+		LOG_ERR("Failed to setup thermal ADC (%d)", ret);
+		goto unlock;
+	}
+
+	ret = adc_sequence_init_dt(&config->adc, &sequence);
+	if (ret < 0) {
+		LOG_ERR("Failed to init thermal ADC sequence (%d)", ret);
+		goto unlock;
+	}
+
+	ret = adc_read(config->adc.dev, &sequence);
+	if (ret < 0) {
+		LOG_ERR("Failed to read thermal ADC (%d)", ret);
+		goto unlock;
+	}
+
+	data->val = buf;
+
+	/* Try to convert to mV if supported, leave as raw otherwise */
+	adc_raw_to_millivolts_dt(&config->adc, &data->val);
+
+unlock:
+	k_mutex_unlock(&data->mutex);
+
+	return ret;
+}
+
+static int adc_thermistor_channel_get(const struct device *dev, enum sensor_channel chan,
+				      struct sensor_value *val)
+{
+	const struct adc_thermistor_config *config = dev->config;
+	struct adc_thermistor_data *data = dev->data;
+	int temp_val = data->val;
+
+	/* Allow getting using ambient or die temperature channel */
+	switch (chan) {
+	case SENSOR_CHAN_AMBIENT_TEMP:
```

Comment:
Why here ALL/ANY is not used?

Reply:
This is common with how other drivers are used. `fetch` is used first to get ALL, while `get` has to be for a single channel.